crates/analytics/src/auth_events/metrics.rs

```diff
@@ -0,0 +1,114 @@
+use api_models::analytics::{
+    auth_events::{AuthEventMetrics, AuthEventMetricsBucketIdentifier},
+    Granularity, TimeRange,
+};
+use time::PrimitiveDateTime;
+
+use crate::{
+    query::{Aggregate, GroupByClause, ToSql, Window},
+    types::{AnalyticsCollection, AnalyticsDataSource, LoadRow, MetricsResult},
+};
+
+mod authentication_attempt_count;
+mod authentication_success_count;
+mod challenge_attempt_count;
+mod challenge_flow_count;
+mod challenge_success_count;
+mod frictionless_flow_count;
+mod three_ds_sdk_count;
+
+use authentication_attempt_count::AuthenticationAttemptCount;
+use authentication_success_count::AuthenticationSuccessCount;
+use challenge_attempt_count::ChallengeAttemptCount;
+use challenge_flow_count::ChallengeFlowCount;
+use challenge_success_count::ChallengeSuccessCount;
+use frictionless_flow_count::FrictionlessFlowCount;
+use three_ds_sdk_count::ThreeDsSdkCount;
+
+#[derive(Debug, PartialEq, Eq, serde::Deserialize)]
+pub struct AuthEventMetricRow {
+    pub total: Option<bigdecimal::BigDecimal>,
+    pub count: Option<i64>,
+    pub time_bucket: Option<String>,
+    pub payment_method: Option<String>,
+    pub platform: Option<String>,
+    pub browser_name: Option<String>,
+    pub source: Option<String>,
+    pub component: Option<String>,
+    pub payment_experience: Option<String>,
+}
```

Comment:
What is this row backed by?
the fields here seem similar to SDK events?

Reply:
Forgot to update this